### PR TITLE
EUI-6754 CAA unassigned cases reset filter

### PR DESCRIPTION
**Pull request by johnbenjamin-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-6754
https://tools.hmcts.net/jira/browse/EUI-6770

### Change description ###
CAA Unassigned cases - Reset filter selection (for Unassigned cases view only)
CAA Assigned and Unassigned cases - Persist filter values entered within session 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by tarikdogan-hmcts — APPROVED** [review]

File: src/caa-cases/components/caa-filter/caa-filter.component.ts [hunk 1]
```diff
@@ -91,10 +90,30 @@ export class CaaFilterComponent implements OnInit, OnChanges, OnDestroy {
...
         this.filteredAndGroupedUsers = filteredAndGroupedUsers;
       });
+      this.initialiseFilterValuesFromSessionState();
+    }
+  }
+
+  public initialiseFilterValuesFromSessionState(): void {
+    if (this.sessionStateValue) {
+      // Set the case reference number input box value if present in session state
+      const caseReferenceNumber = this.sessionStateValue.caseReferenceNumber && this.sessionStateValue.caseReferenceNumber;
```
Comment (Daniel-Lam) [inline src/caa-cases/components/caa-filter/caa-filter.component.ts:106]: @johnbenjamin-hmcts - looks like a duplicate check for `this.sessionStateValue.caseReferenceNumber`?

File: src/caa-cases/components/caa-filter/caa-filter.component.ts [hunk 2]
```diff
@@ -91,10 +90,30 @@ export class CaaFilterComponent implements OnInit, OnChanges, OnDestroy {
...
+      this.caaFormGroup.get(this.caseRefFormControl).setValue(caseReferenceNumber);
+      // Set the assignee name input box value if present in session state
+      if (this.caaCasesPageType === CaaCasesPageType.AssignedCases) {
+        const assigneeName = this.sessionStateValue.assigneeName && this.sessionStateValue.assigneeName;
```
Comment (Daniel-Lam) [inline src/caa-cases/components/caa-filter/caa-filter.component.ts:110]: @johnbenjamin-hmcts - duplicate check for `this.sessionStateValue.assigneeName`

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts [hunk 3]
```diff
@@ -261,9 +266,57 @@ export class CaaCasesComponent implements OnInit {
...
+  public removeSessionState(key: string): void {
+    this.service.removeSessionState(key);
+  }
+
+  public retrieveSessionState(): void {
+    this.sessionStateValue = this.service.retrieveSessionState(this.caaCasesPageType);
+    if (this.sessionStateValue) {
+      this.selectedFilterType = this.sessionStateValue.filterType ? this.sessionStateValue.filterType : null;
+      if (this.selectedFilterType) {
+        const caseReferenceNumber = this.sessionStateValue.caseReferenceNumber && this.sessionStateValue.caseReferenceNumber;
```
Comment (Daniel-Lam) [inline src/caa-cases/containers/caa-cases/caa-cases.component.ts:289]: @johnbenjamin-hmcts - duplicate `this.sessionStateValue.caseReferenceNumber` check

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts [hunk 4]
```diff
@@ -261,9 +266,57 @@ export class CaaCasesComponent implements OnInit {
...
+        const assigneeName = this.sessionStateValue.assigneeName && this.sessionStateValue.assigneeName;
```
Comment (Daniel-Lam) [inline src/caa-cases/containers/caa-cases/caa-cases.component.ts:290]: @johnbenjamin-hmcts - duplicate `this.sessionStateValue.assigneeName` check

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts [hunk 5]
```diff
@@ -261,9 +266,57 @@ export class CaaCasesComponent implements OnInit {
...
+          }
+        }
+        this.toggleFilterSection();
+      }
+    }
+  }
+
+  public storeSessionState(): void {
+    const sessionStateValue = this.service.retrieveSessionState(this.caaCasesPageType);
+    const caseReferenceNumber = sessionStateValue && sessionStateValue.caseReferenceNumber && sessionStateValue.caseReferenceNumber;
```
Comment (Daniel-Lam) [inline src/caa-cases/containers/caa-cases/caa-cases.component.ts:307]: @johnbenjamin-hmcts - duplicate `sessionStateValue.caseReferenceNumber` check

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts [hunk 6]
```diff
@@ -261,9 +266,57 @@ export class CaaCasesComponent implements OnInit {
...
+    const assigneeName = sessionStateValue && sessionStateValue.assigneeName && sessionStateValue.assigneeName;
```
Comment (Daniel-Lam) [inline src/caa-cases/containers/caa-cases/caa-cases.component.ts:308]: @johnbenjamin-hmcts - duplicate `sessionStateValue.assigneeName` check

**Review by Daniel-Lam — CHANGES_REQUESTED** [review]
@johnbenjamin-hmcts - a few comments to address. Also, I might not have spotted it, but did you test for the scenario where the user enters a Case Reference on the Unassigned Cases filter, then switches to the Assigned Cases filter, and vice versa? I believe it was discussed that the Case Reference number should be cleared in both instances.

**Comment by johnbenjamin-hmcts** [comment]
> @Daniel-Lam - There will be two session storage items, one for assigned cases and the other for unassigned cases and they are mutually exclusive. So, the case reference entered on both screens will be retained.

File: src/caa-cases/components/caa-filter/caa-filter.component.ts (hunk 1 above)
Comment (johnbenjamin-hmcts) [reply src/caa-cases/components/caa-filter/caa-filter.component.ts:106]: @Daniel-Lam - I used this statement to get the `caseReferenceNumber` only if it is defined.

File: src/caa-cases/components/caa-filter/caa-filter.component.ts (hunk 2 above)
Comment (johnbenjamin-hmcts) [reply src/caa-cases/components/caa-filter/caa-filter.component.ts:110]: @Daniel-Lam - I used this statement to get the `assignee` only if it is defined.

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts (hunk 3 above)
Comment (johnbenjamin-hmcts) [reply src/caa-cases/containers/caa-cases/caa-cases.component.ts:289]: @Daniel-Lam - I used this statement to get the `caseReferenceNumber` only if it is defined.

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts (hunk 5 above)
Comment (johnbenjamin-hmcts) [reply src/caa-cases/containers/caa-cases/caa-cases.component.ts:307]: @Daniel-Lam - I used this statement to get the `caseReferenceNumber` only if it is defined.

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts (hunk 6 above)
Comment (johnbenjamin-hmcts) [reply src/caa-cases/containers/caa-cases/caa-cases.component.ts:308]: @Daniel-Lam - I used this statement to get the `assignee name` only if it is defined.

File: src/caa-cases/containers/caa-cases/caa-cases.component.ts (hunk 4 above)
Comment (johnbenjamin-hmcts) [reply src/caa-cases/containers/caa-cases/caa-cases.component.ts:290]: @Daniel-Lam - I used this statement to get the `assignee name` only if it is defined.

**Review by Daniel-Lam — APPROVED** [review]

**Pull request merged**